caas/kubernetes/clientconfig/k8s_test.go

```diff
@@ -172,6 +173,67 @@ func (s *k8sConfigSuite) assertSingleConfig(c *gc.C) {
 		})
 }
 
+func StdinMockContextManager(content []byte, callBack func()) {
```

Comment:
I wonder if we have a mock without actually writing a file.... There are some precedences in other places where files are used and tested.

Reply:
I could't find one from our repo. Here is mock `os.Stdin`